### DISCUSSION TITLE
more get_node() sugar, please!

### DISCUSSION THREAD

**Discussion by BaddRadish** [godotengine/godot-proposals#5474] [Engine Core]
currently `$child` is sugar for `get_node("child")`. great. love it.
i think we should consider expanding this...

(1) GENERAL NODE ACCESS
sometimes we need to access the child of _a node that is stored in a variable_, instead of the current script's context.
so instead of `get_node("child")`, sometimes we need `my_var.get_node("child")`.
currently there's no sugar for this.
but we should add some:
`my_var.$child.attribute`
- it feels consistent with both `$` behavior and `.` behavior.
- its simple.
- the highlighter already correctly highlights this syntax.

(2) PARENT NODE ACCESS
to access the parent or siblings right now we need quotes: `$"../sibling"`.
as far as i can tell this is the only time you _need_ to put quotes around the path.
i cant imagine it would be _too_ hard to make `$../sibling` work.
i suppose the problem would be `.attribute`.
anyway some kind of sugar for `.get_parent()` would be nice.
`$".."` just _feels wrong_.
- as far as i can tell this is the only time you _need_ to put quotes around the path, so needing quotes here is inconsistent with all other uses of `get_node()`.
- i cant imagine it would be _too_ hard to make `$../sibling` work.

i suppose the problem would be `.attribute`.
anyway some kind of sugar for `.get_parent()` would be nice.

(3) $ FOR NON LITERAL STRINGS
suppose i wanted to dynamically generate a path. i could do that with `get_node("path"+my_str)`.
but if i want to use `$` i can only use literals.
how about `$("path"+my_str)`?
- kind of like how we have to add quotes back in for the special case of `$".."`, in this special case we also have to add the parens back in.
- it still works the same as `get_node()`, so its not complicating the `$` to `get_node()` relationship at all.

i'm not _too_ worried about this one, just an idea.

thanks for your time and input!

**Comment by Calinou** [comment]
Have you seen scene-unique nodes (`%Something` in 4.0, or `get_node("%Something")` in 3.5)? This allows you to access nodes within a scene regardless of their actual path by passing only their name after the `%` symbol. You can create those by right-clicking a node in the Scene tree dock and choosing **Access As Scene Unique Node**.

**Reply by BaddRadish** [reply]
yes i have seen this. thank you!

**Comment by Mickeon** [comment]
See https://github.com/godotengine/godot-proposals/issues/1776#issuecomment-722388271.

Also see https://github.com/godotengine/godot-proposals/issues/5064 and https://github.com/godotengine/godot-proposals/issues/5064 for more... possibly savoury, sugary related Proposals.

**Reply by BaddRadish** [reply]
added my two cents to those
